src/app.py

```diff
@@ -83,40 +82,41 @@ def return_datasets(self):
                 dataset_list.append(dataset)
         return dataset_list
 
+
 def process_files(dataset, shared_list, finished_list, config_info):
     process_name = multiprocessing.current_process().name
     url = dataset['url']
     data_type = dataset['type']
     data_sub_type = dataset['subtype']
     filename = dataset['filename']
-    save_path = 'files'
+    save_path = '/usr/src/app/tmp'
 
-    logger.debug('{} is processing {}'.format(process_name, dataset))
-    
     if dataset['status'] == 'active':
         if url not in shared_list and url not in finished_list:
             shared_list.append(url)
             download(process_name, url, filename, save_path)
             shared_list.remove(url)
-            decompress(process_name, filename, save_path)
```

Comment:
The decompress function will check for the ends of filenames and then either decompress or skip decompression if it's not required. It serves as both a "checker" and also a dispatch to decompression. If it's OK, I'd like to keep it instead of adding lots of .endswith('abc') since it already does that.

Reply:
without this bit it gave a 'NoneType' error...maybe the decompress/skip compress isn't working quite right?

Reply:
Ah, OK. Yes, it must be buggy then. Can I take a look at your branch tomorrow?